packages/workspace/src/browser/workspace-service.ts

```diff
@@ -315,49 +319,53 @@ export class WorkspaceService implements FrontendApplicationContribution {
     /**
      * Removes root folder(s) from workspace.
      */
-    async removeRoots(uris: URI[]): Promise<void> {
-        if (!this.opened) {
-            throw new Error('Folder cannot be removed as there is no active folder in the current workspace.');
-        }
-        if (this._workspace) {
-            const workspaceData = await this.getWorkspaceDataFromFile();
-            this._workspace = await this.writeWorkspaceFile(this._workspace,
-                WorkspaceData.buildWorkspaceData(
-                    this._roots.filter(root => uris.findIndex(u => u.toString() === root.uri) < 0),
-                    workspaceData!.settings
-                )
-            );
-        }
+    removeRoots(uris: URI[]): Promise<void> {
+        return this.writeQueue.push(async () => {
+            if (!this.opened) {
```

Comment:
it would be better to extract such protected unsafe methods in order to reuse them and have less changes